front/src/modules/Items/ItemList.tsx

```diff
@@ -1,57 +1,117 @@
-import {Item} from "./components/Item";
-import {FixedSizeList as List} from "react-window";
+import { WarehouseEntryWithPath } from "../../../../back/src/modules/warehouse";
+import { Item } from "./components/Item";
 
 export function ItemList(props: {
   list: WarehouseEntryWithPath[];
   onDeleteItem: (item: WarehouseEntryWithPath) => void;
   onUpdateItem: (item: WarehouseEntryWithPath) => void;
-  onCreateItem: (item: {name: string}) => void;
+  onCreateItem: (item: { name: string }) => void;
   onOpenItem: (item: WarehouseEntryWithPath) => void;
   onStartCutting: (item: WarehouseEntryWithPath) => void;
-  cutting: null | {item: WarehouseEntryWithPath};
+  cutting: null | { item: WarehouseEntryWithPath };
   isSearch: boolean;
 }) {
   const [newItemName, setNewItemName] = useState("");
+  const [slicedArray, setSlicedArray] = useState<WarehouseEntryWithPath[]>([]);
+  const [refState, setRefState] = useState(false);
+  const [isObserving, setIsObserving] = useState(false);
+  const observerRef = useRef<IntersectionObserver | null>(null);
+
+  const listLength = props.list.length;
+  const currentIndexRef = useRef(20);
+  const oldIndexRef = useRef(0);
+
+  useEffect(() => {
+    if (slicedArray.length > listLength) {
+      setSlicedArray(props.list.slice(0, currentIndexRef.current));
+      currentIndexRef.current = Math.min(20, listLength);
+      oldIndexRef.current = 0;
+      setRefState(false);
+    }
+    if (refState) {
+      if (slicedArray.length < listLength) {
+        const currentIndex = currentIndexRef.current;
+        const oldIndex = oldIndexRef.current;
+
+        setSlicedArray(prev => [
+          ...prev,
+          ...props.list.slice(oldIndex, currentIndex),
+        ]);
+
+        oldIndexRef.current = currentIndex;
+        currentIndexRef.current = Math.min(currentIndex + 10, listLength);
+
+        setRefState(false);
+      }
+    }
+  }, [listLength, refState, props.list, slicedArray]);
 
   function handleCreateItem() {
-    props.onCreateItem({name: newItemName});
+    props.onCreateItem({ name: newItemName });
     setNewItemName("");
   }
 
-  const Row = ({index, style}: {index: number; style: React.CSSProperties}) => {
-    const item = props.list[index];
-    return (
-      <div style={style}>
-        <Item
-          key={item.id}
-          isSearch={props.isSearch}
-          item={item}
-          onDelete={() => props.onDeleteItem(item)}
-          onGoForward={() => props.onOpenItem(item)}
-          onEdit={props.onUpdateItem}
-          onCutStart={() => props.onStartCutting(item)}
-          cutting={props.cutting}
-        />
-      </div>
-    );
-  };
+  useEffect(() => {
+    const options = {
+      root: document.querySelector("#root"),
+      rootMargin: "0px",
+      threshold: 0.7,
+    };
+
+    if (!observerRef.current) {
+      observerRef.current = new IntersectionObserver(entries => {
+        const [entry] = entries;
+        if (entry.isIntersecting && !isObserving) {
```

Comment:
if `entries` are empty, `entry` will be `undefined` and `isIntersecting` would be accessed on `undefined` which will crash the app.